games/naval_battle.c

```diff
@@ -0,0 +1,802 @@
+/**
+ * @file
+ * @author [Carlos Rafael](https://github.com/CarlosZoft)
+ * @author [Herick Lima](https://github.com/hericklima22)
+ * @brief [naval_battle](https://en.wikipedia.org/wiki/Battleship_(game))
+ * implementation in C using only the stdio.h library.
+ * @details  Naval battle is a game, to be played by two people. It consists of
+ * knocking down the enemy ship, through shots , when hit the ship is
+ * revealed with the respective number of its size. Example: size 3 = 3 3 3 on
+ * the board.
+ */
+#include <stdio.h>
+
+/**
+ * Function validEntryLineColumn
+ * Responsible for validating entries, for positioning boats
+ * @param line -> matrix row
+ * @param column -> matrix column
+ * @returns -> validates row and column entry of the board
+ */
+int validEntryLineColumn(int line, char column)
+{
+    if ((line >= 1 && line <= 10) && (column >= 65 && column <= 74))
+    {
+        return 1;
+    }
+
+    else
+        return 0;
+}
+/**
+ * Function validatePosition
+ * Responsible for checking if the position can receive the boat.
+ * @param -> mat board
+ * @param -> boat boat
+ * @param -> line matrix row
+ * @param -> column matrix column
+ * @returns -> checks if the position is valid
+ */
+int validatePosition(int mat[10][10], int boat, int line, int column,
+                     char guide)
+{
+    int cont = 0;
+    int i, j;
+
+    if (boat < 1 || boat > 3)
+        return 0;
+    if (guide != 'H' && guide != 'V')
+        return 0;
+    if ((line < 0 || line > 9) || (column < 0 || column > 9))
+        return 0;
+
+    if (guide == 'H')
+    {
+        if ((10 - column) < boat)
+            return 0;
+
+        else
+        {
+            for (j = column; j < (column + boat); j++)
+            {
+                if (mat[line][j] == 0)
+                    cont++;
+            }
+        }
+    }
+
+    if (guide == 'V')
+    {
+        if ((10 - line) < boat)
+            return 0;
+
+        else
+        {
+            for (i = line; i < (line + boat); i++)
+            {
+                if (mat[i][column] == 0)
+                    cont++;
+            }
+        }
+    }
+
+    if (cont == boat)
+    {
+        return 1;
+    }
+    else
+    {
+        return 0;
+    }
+}
+/**
+ * Function canShoot
+ * Responsible to verify that it is a valid position to shoot
+ * @param mat -> board
+ * @param line -> matrix row
+ * @param column -> matrix column
+ * @returns -> checks if the position is valid for shooting
+ */
+
+int canShoot(int mat[10][10], int line, int column)
+{
+    if (mat[line][column] == -2 || mat[line][column] == 10 ||
+        mat[line][column] == 20 || mat[line][column] == 30 ||
+        mat[line][column] == 50)
+    {
+        return 0;
+    }
+
+    else
+    {
+        return 1;
+    }
+}
+/**
+ * Function positionBoat
+ * Responsible for placing the boats on the board, according to the size.
+ * @param mat -> board
+ * @param boat -> boat
+ * @returns(void) -> position the boat on the board
```

Comment:
Please check out other parts of the code for this.
```suggestion
 * @param mat board
 * @param boat boat
 * @returns(void) position the boat on the board
```